gmsh_interop/runner.py

```diff
@@ -149,12 +159,18 @@ def get_gmsh_version_from_string(output: str) -> Version | None:
 
 
 class GmshRunner:
-    def __init__(self, source, dimensions=None, order=None,
-            incomplete_elements=None, other_options=(),
-            extension="geo", gmsh_executable="gmsh",
-            output_file_name=None,
-            target_unit=None,
-            save_tmp_files_in=None):
+    def __init__(
+            self,
+            source: str | ScriptSource | FileSource | ScriptWithFilesSource,
+            dimensions: int | None = None,
+            order: int | None = None,
+            incomplete_elements: Literal[0, 1] | None = None,
```

Comment:
How come that's not a bool? Is `Literal[0, 1]` compatible with `bool`?

Reply:
I made it a bool in https://github.com/inducer/gmsh_interop/compare/cb94616ec703f23f8c6789ed451cac3964950637..0f4f6ee41572837ba45b3f4aed6a443efcd1fd80.

I wasn't quite sure what type this should be from the Gmsh docs, where it says that the default value for `Mesh.SecondOrderIncomplete` is 0. Guess it's supposed to be a boolean flag? 
https://gmsh.info/doc/texinfo/gmsh.html#Mesh-options

Reply:
I agree, it's not totally clear from the gmsh docs, but it seems to be a yes/no thing. So `bool` is probably the appropriate translation.